tasks/main.yml

```diff
@@ -17,26 +12,90 @@
       {{ ansible_facts['distribution'] ~ '_' ~
       ansible_facts['distribution_version'] }}.yml
   vars:
-    __template_vars_file: "{{ role_path }}/vars/{{ item }}"
-  when: __template_vars_file is file
+    __mssql_vars_file: "{{ role_path }}/vars/{{ item }}"
+  when: __mssql_vars_file is file
+
+- name: "Verify that the user accepts EULA variables"
+  assert:
+    that:
+      - mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula
+      - mssql_accept_microsoft_cli_utilities_for_sql_server_eula
+      - mssql_accept_microsoft_sql_server_2019_standard_eula
+    fail_msg:
+      - "You must accept EULA by setting the following variables to true:"
+      - "mssql_accept_microsoft_odbc_driver_17_for_sql_server_eula"
+      - "mssql_accept_microsoft_cli_utilities_for_sql_server_eula"
+      - "mssql_accept_microsoft_sql_server_2019_standard_eula"
+  register: __mssql_eula_evaluation
+
+- name: Configure the Microsoft SQL Server 2019 repository
+  yum_repository:
+    name: packages-microsoft-com-mssql-server-2019-gdr
+    description: Microsoft SQL Server 2019
+    baseurl: "{{ __mssql_server_repository }}"
+    gpgkey: https://packages.microsoft.com/keys/microsoft.asc
+    gpgcheck: true
+
+- name: Ensure that the Microsoft SQL Server package is installed
+  package:
+    name: "{{ __mssql_server_packages }}"
+    state: present
+
+- name: Configure the Microsoft SQL Server Tools repository
+  yum_repository:
+    name: packages-microsoft-com-prod
+    description: Microsoft SQL Server Tools
+    baseurl: "{{ __mssql_client_repository }}"
+    gpgcheck: true
 
-# Examples of some tasks:
-- name: Ensure required packages are installed
+- name: Ensure that SQL Server client tools are installed
   package:
-    name: "{{ __template_packages }}"
+    name: "{{ __mssql_client_packages }}"
     state: present
+  environment:
+    - ACCEPT_EULA: Y
+
+- name: Gather system services facts
+  service_facts:
+
+- name: Set up MS SQL Server
+  command: /opt/mssql/bin/mssql-conf -n setup accept-eula
+  environment:
+    - ACCEPT_EULA: "Y"
+    - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+    - MSSQL_PID: "{{ mssql_edition }}"
+    - MSSQL_TCP_PORT: "{{ mssql_db_port }}"
+  when: |
+    "running" not in
+    ansible_facts["services"]["mssql-server.service"]["state"]
+
+- name: Input the {{ mssql_db_sql_file }} file to MSSQL Server
+  when: mssql_db_sql_file is defined
+  block:
+    - name: Create a tempfile on the host
+      tempfile:
+        state: file
+      register: __mssql_sql_tempfile
+
+    - name: Copy the {{ mssql_db_sql_file }} file to the host
+      copy:
+        src: "{{ mssql_db_sql_file }}"
+        dest: "{{ __mssql_sql_tempfile.path }}"
+        mode: preserve
+
+    - name: Input the {{ mssql_db_sql_file }} file with the sqlcmd command
+      command: |
+        /opt/mssql-tools/bin/sqlcmd -U sa -P {{ mssql_password }}
```

Comment:
when you pass mssql_password to the command line it needs to be quoted.  I'm not sure what are the rules for the `command` module e.g. what happens if there is a space character in mssql_password?  does ansible/yaml parse the string value and tokenize it before or after expanding the `{{ mssql_password }}`?  You don't have to worry about the tempfile path because it is guaranteed to be shell safe.

Reply:
I tested with a password containing a space `p@55 w0rD` and it indeed failed. Then I added the quotes around the mssql_password var here and then the `p@55 w0rD` password worked. So quotes are required, and with quotes space within the password works fine.

Reply:
Also, if the `mssql_password` var is not defined, the `Set up MS SQL Server` task fails with the following output:
`fatal: [/home/spetrosi/Downloads/rhel-guest-image-7.9-30.x86_64.qcow2]: FAILED! => {"msg": "The field 'environment' has an invalid value, which includes an undefined variable. The error was: 'mssql_password' is undefined\n\nThe error appears to be in '/home/spetrosi/Documents/work/linux-system-roles/mssql/tasks/main.yml': line 61, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Set up MS SQL Server\n  ^ here\n"}`
Is that fine?

Reply:
.done.